superdesk/attachments.py

```diff
@@ -73,6 +74,23 @@ def is_attachment_public(attachment):
     return not attachment.get('internal')
 
 
+def get_attachment_public_url(attachment: Dict[str, Any]) -> Optional[str]:
+    """Returns the file url for the attachment provided
+
+    :param dict attachment: The attachment to get the file URL
+    :rtype: str
+    :return: None if the attachment is not public, otherwise the public URL to the file
+    """
+
+    if attachment.get('attachment'):  # retrieve object reference
+        attachment = superdesk.get_resource_service('attachments').find_one(req=None, _id=attachment['attachment'])
+
+    if attachment.get('internal'):
+        return None
+
+    return current_app.media.url_for_external(attachment['media'], RESOURCE)
```

Comment:
Is there any chance that `media` not in `attachment`?

Reply:
Attachments **should** have a `media` field.
I have placed a check and logged a warning if this is not the case